GPU/Common/ShaderUniforms.cpp

```diff
@@ -23,6 +23,28 @@ static void ConvertProjMatrixToD3D11(Matrix4x4 &in) {
 	in.translateAndScale(trans, scale);
 }
 
+void ComputeGuardband(float gb[4], float zmin) {
+	float vpWidth = fabsf(gstate_c.vpWidth);
+	float vpHeight = fabsf(gstate_c.vpHeight);
+	// Avoid bad values during initialization. Doubt these are really needed.
+	if (vpWidth == 0.0)
+		vpWidth = 480;
+	if (vpHeight == 0.0)
+		vpHeight = 272;
+
+	// We assume a symmetric guardband, even though it's not entirely correct to do so - but nearly everything does it
+	// this way and we have space for the NAN in the uniform.
+	// We also assume that everything behind the near clipping plane gets clipped and will thus not in reality
+	// exceed the guardband. This is a bit rough but should be ok.
+	float offsetX = gstate.getOffsetX();
+	float offsetY = gstate.getOffsetY();
```

Comment:
Well, in theory if the game has the viewport say at the far left of the 4096x4096 square, it could draw something very wide only to the right that we'd now cull but shouldn't. I don't think that's much of a concern though.